addons/vehicles/functions/fnc_speedLimiter.sqf

```diff
@@ -38,21 +39,25 @@ GVAR(speedLimit) = speed _vehicle max 5;
         private _uavControll = UAVControl _vehicle;
         if ((_uavControll select 0) != _driver || _uavControll select 1 != "DRIVER") then {
             GVAR(isSpeedLimiter) = false;
+            _vehicle setCruiseControl [0, false];
         };
     } else {
         if (_driver != driver _vehicle) then {
             GVAR(isSpeedLimiter) = false;
+            _vehicle setCruiseControl [0, false];
         };
     };
 
     if (!GVAR(isSpeedLimiter)) exitWith {
+        _vehicle setCruiseControl [0, false];
         [_idPFH] call CBA_fnc_removePerFrameHandler;
     };
 
+    _vehicle setCruiseControl [GVAR(speedLimit), false];
```

Comment:
Don't do this!
setCruiseControl script command resets the thrust PID Controller.

If you reset it every frame you'll completely mess up the auto thrust by resetting the PID basically every simulation cycle.

Only do this if you turn on/off or if the speedLimit actually changes.
Engine doesn't check that provided speed is actually different from current setpoint, it just applies what you give it.

Reply:
If you reset the PID every frame, that will likely also cause your mentioned issues with going too fast downhill.